### ISSUE TITLE
Use UUIDs instead of Relay IDs

### ISSUE THREAD

**Issue by nirshubz** [graphql-python/graphene#243]
Hey Guys,

Awesome library 👍 
We are looking for some best practises regarding the use of Relay IDs together with UUIDs.

**Our Setup**
We are using Django Graphene with two different setups:
1. Redux & GraphQL (no Relay)
2. Relay

We are using a postgresql for DB. Our django models generate UUIDs for all our entities. We really like the UUIDs :)

Based on the following issue https://github.com/facebook/relay/issues/191 and the fact that we want the client to use a single ID throughout the app, we have decided that the client side should only know about the Relay ID. 

**The Problem**

This raises three issues:
1.  Mutations and advanced resolves with foreign keys need to be translated from Relay IDs to our UUIDs. 
2. More importantly, it makes it much harder to debug ids all the way from client to server.
3. It makes it impossible to use non graphql APIs.

**Current Solution**
We have written a middleware that solves problem (1), by translating IDs to our UUIDs. 
However, this does not solve (2) and (3), and we still feel our solution is kind of patchy :)  

**Questions**
1. Can we use our UUIDs instead of the base64 IDs generated by graphene? Would it still work fine with Relay?
2.Any other suggestions or other best practises we should use?

Thanks!
Nir

**Comment by syrusakbary** [comment]
Hi @	nirshubz, it's possible to use the `UUID`s as long as you can map from a UUID to a type/model.
So, given a UUID get the corresponding model.

With the current implementation this is not as easy to achieve but with the upcoming version `1.0` it will be trivial:
https://github.com/graphql-python/graphene/blob/1.0/graphene/relay/tests/test_node_custom.py

**Comment by nirshubz** [comment]
Got you. Thanks for the answer.
Our UUID is generated by an external system, and is completely random, so it would be kind of hard to follow that logic.

Do you think it would be possible to somehow figure out the mapping from the query when using Django? Any other suggestions?   

Thanks :)

**Comment by syrusakbary** [comment]
A quick update on this issue!
The `1.0.dev` version is live on PyPI, I recommend you to give it a try having a similar approach to this: https://github.com/graphql-python/graphene/blob/master/graphene/relay/tests/test_node_custom.py

Usually for mapping this random id's to type+id is useful to have a "utility table" that stores what is the table/type corresponding to each "UUID".

Hope that helps!
